pr_labeler: add warning for porting_guides changes

gotmax23 opened the pull request:
This adds a warning message when PRs are created that edit
porting_guides by someone outside of the Release Management WG. These
files are automatically generated during the ansible release process and
should not be modified.

Fixes: https://github.com/ansible/ansible-documentation/issues/503

gotmax23 committed:
pr_labeler: improve create_boilerplate_comment logging

gotmax23 committed:
pr_labeler: add --force-process-closed flag

gotmax23 committed:
pr_labeler: add warning for porting_guides changes

This adds a warning message when PRs are created that edit
porting_guides by someone outside of the Release Management WG. These
files are automatically generated during the ansible release process and
should not be modified.

Fixes: https://github.com/ansible/ansible-documentation/issues/503

felixfontein commented on hacking/pr_labeler/label.py:
```diff
@@ -225,6 +236,21 @@ def no_body_nag(ctx: IssueOrPrCtx) -> None:
+        return
+    matches: list[str] = []
+    for file in ctx.pr.get_files():
+        if file.filename.startswith("docs/docsite/rst/porting_guides/"):
```
Files starting with `porting_guide_(core|base)` shouldn't be considered.

Also porting guides for older Ansible versions will never be updated again, so for them no warning should be printed as well.

gotmax23 replied:
I don't think we should update old porting_guides either. They are historical documents that should be left alone, and we should leave the entries from collections' changelogs alone in the same way we don't patch their code.

samccann replied:
We did just update some of the old stuff to fix spelling errors. I can't  think of an example where we updated an EOL porting guide for anything significant, tho.

felixfontein replied:
I had to fix a broken reference in one of them recently to fix CI in this repo.

samccann replied:
hmm... that said... are we even able to regenerate older porting guides if something dramatic is noticed and needs a fix? Would this label just add a comment but still allow us to merge some critical PR if necessary for an EOL porting guide?

gotmax23 replied:
This doesn't block anything. It simply adds a comment. We can always ignore it, but at least the warning is prominent.

samccann replied:
cool thanks

samccann commented on hacking/pr_labeler/data/porting_guide_changes.md:
```diff
@@ -0,0 +1,8 @@
+The following files are automatically generated and should not be modified outside of the ansible release process:
```
```suggestion
The following files are automatically generated and should not be modified outside of the Ansible release process:
```

samccann commented on hacking/pr_labeler/data/porting_guide_changes.md:
```diff
@@ -0,0 +1,8 @@
+- {{ file }}
+{% endfor %}
+
+Please double check your changes.
```
```suggestion
Please double-check your changes.
```

samccann commented on hacking/pr_labeler/label.py:
```diff
@@ -26,6 +26,14 @@
+    "rooftopcellist",
+    "webknjaz",
+}
 HERE = Path(__file__).resolve().parent
```
should we have some kind of team like the steering committee has so we don't have to edit this list directly if release managers come and go?

gotmax23 replied:
We could create an @ansible/release-management-wg (release-management<strong>-wg</strong> is to disambiguate @ansible/ansible-release-managers) group, but group members aren´t public, so we wouldn´t be able to use that data here.

gotmax23 replied:
> group members aren´t public, so we wouldn´t be able to use that data here.

Once we set up the authentication methods proposed in https://github.com/ansible/ansible-documentation/issues/382, we'll be able to access this data.

gotmax23 committed:
pr_labeler: use @release-management-wg team for porting_guide check

Instead of hardcoding the list of release managers, we can use the
Github API to retrieve the members of the
`@ansible/release-management-wg` team.

gotmax23 committed:
pr_labeler: exempt bots from porting_guide check

For example, patchback is not a release manager, but we still want it to
backport Porting Guide PRs.

gotmax23 committed:
pr_labeler: improve porting_guide_changes template wording

Co-authored-by: Sandra McCann <samccann@redhat.com>

gotmax23 commented:
Okay, this should be ready for another review.

The list of release managers are no longer hardcoded. Now that we have the Github app token set up, we can retrieve the list of release managers from @ansible/release-management-wg via the Github API :tada:.

This no longer applies to core porting guides; warnings are only printed for changes to community porting guides. There is [deliberately no distinction](https://github.com/ansible/ansible-documentation/pull/514#discussion_r1346339190) between current and EOL changelogs.

samccann reviewed (approved):
Can't really comment on the code but it all LGTM.

gotmax23 commented:
Thanks for the review, @samccann!

gotmax23 commented:
I just rebased on top of `devel`.

gotmax23 commented:
Note to self: I tagged this PR with `merge_commit` so it should be applied with a merge commit and *not* `Squash and merge`.
